src/storage/Storage.ts

```diff
@@ -4,17 +4,22 @@ import { ExitChannelBlockStorage } from "./ExitChannelBlockStorage";
 import { StateSnapshotStorage } from "./StateSnapshotStorage";
 import { ExitPointsStorage } from "./ExitPointsStorage";
 import { Block, BlockCoordinates, StateSnapshot } from "@/models";
-import { createDeepCopyProxy } from "@/utils";
+import { deepCopyProxy } from "@/utils";
 
 export class Storage {
-    public readonly blocks = new BlockStorage();
-    public readonly joinChannelBlocks = new JoinChannelBlockStorage();
-    public readonly exitChannelBlocks = new ExitChannelBlockStorage();
-    public readonly stateSnapshots = new StateSnapshotStorage();
-    public readonly exitPoints = new ExitPointsStorage();
+    public readonly blocks: BlockStorage;
+    public readonly joinChannelBlocks: JoinChannelBlockStorage;
+    public readonly exitChannelBlocks: ExitChannelBlockStorage;
+    public readonly stateSnapshots: StateSnapshotStorage;
+    public readonly exitPoints: ExitPointsStorage;
 
     constructor() {
-        return createDeepCopyProxy(this);
+        this.blocks = deepCopyProxy(new BlockStorage());
+        this.joinChannelBlocks = deepCopyProxy(new JoinChannelBlockStorage());
+        this.exitChannelBlocks = deepCopyProxy(new ExitChannelBlockStorage());
+        this.stateSnapshots = deepCopyProxy(new StateSnapshotStorage());
+        this.exitPoints = deepCopyProxy(new ExitPointsStorage());
+        return deepCopyProxy(this);
```

Comment:
Fine for now - Has a drawback when used with helper functions defined in `Storage` it will perform the deep copy again even though calling the function already performed the deep copy - But still better than cascading the proxy recursively and always creating new instances - I'll put an optimization card for this at the end of our backlog - this is enough for now